pkg/image/registryclient/client.go

```diff
@@ -226,6 +244,37 @@ func (c *Context) Repository(ctx context.Context, registry *url.URL, repoName st
 	return NewLimitedRetryRepository(repo, c.Retries, limiter), nil
 }
 
+// RepositoryWithAlternateReference will return the next cached repository and ref, or will gather alternate sources then return first repo, ref, or error
+func (c *Context) RepositoryWithAlternateReference(ctx context.Context, locator imagereference.DockerImageReference, registry *url.URL, insecure bool) (distribution.Repository, imagereference.DockerImageReference, error) {
```

Comment:
Do we need this `registry` argument? Seems like it is not used.

Reply:
probably left over from previous implementation I'll take a look, thanks

Reply:
removed, fixed